### PR TITLE
Visible tracks, Configure Map – Web

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2338

File: map/src/menu/visibletracks/VisibleTracks.jsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                        isLastItem={isLastItem}
+                    />
+                );
+            }
+        });
+        return items;
+    }, [ctx.visibleTracks]);
+
+    function hasVisibleTracks() {
+        return !!(!isEmpty(ctx.visibleTracks?.old) || !isEmpty(ctx.visibleTracks?.new));
```
Comment (RZR-UA) [inline map/src/menu/visibletracks/VisibleTracks.jsx:80]: `!!` seems too redundant as for already boolean expression

**Review by RZR-UA — APPROVED** [review]

**Review by RZR-UA — APPROVED** [review]

File: map/src/context/AppContext.js [hunk 2]
```diff
@@ -83,33 +83,50 @@ async function loadListFiles(
...
+                newSelectedFiles.push(_.indexOf(files, matchingFile));
             }
         });
     }
+    newVisFilesNames.old = newVisFilesNames?.old.splice(-10);
+    newVisFilesNames.old.forEach((name) => {
+        const matchingFile = files.find((f) => f.name === name);
+        if (matchingFile) {
+            oldSelectedFiles.push(_.indexOf(files, matchingFile));
+        }
```
Comment (RZR-UA) [inline map/src/context/AppContext.js:126]: Could you think about turning `files.find + indexOf` into `files.findIndex` ?

File: map/src/context/AppContext.js [hunk 3]
```diff
@@ -119,21 +136,44 @@ async function addOpenedTracks(files, gpxFiles, setGpxFiles, setVisibleTracks) {
...
+        let f = await Utils.getFileData(newGpxFiles[file.name]);
+        const gpxfile = new File([f], file.name, {
+            type: 'text/plain',
+        });
+
+        promises.push(
+            TracksManager.getTrackData(gpxfile).then((track) => {
+                track.name = file.name;
+                Object.keys(track).forEach((t) => {
+                    newGpxFiles[file.name][`${t}`] = track[t];
```
Comment (RZR-UA) [inline map/src/context/AppContext.js:164]: [t]

File: map/src/map/util/TrackLayerProvider.js [hunk 4]
```diff
@@ -13,7 +12,7 @@ export const TEMP_LINE_STYLE = {
     // name: TEMP_LAYER_FLAG, // style.name was not used, instead of actual layer.options.name
 };
 
-function createLayersByTrackData(data, ctx, map) {
+function createLayersByTrackData(data, ctx, map, type = GPX_FILE_TYPE) {
```
Comment (RZR-UA) [inline map/src/map/util/TrackLayerProvider.js:15]: Please change to params to { data, ctx, map, type = GPX_FILE_TYPE }

File: map/src/menu/actions/TrackActions.jsx [hunk 5]
```diff
@@ -53,6 +53,7 @@ const TrackActions = forwardRef(({ track, setDisplayTrack, setOpenActions }, ref
                 <Paper id="se-track-actions" className={styles.actions}>
                     {ctx.gpxFiles[track.name]?.showOnMap ? (
                         <MenuItem
+                            id={'se-hide-from-map-action'}
```
Comment (RZR-UA) [inline map/src/menu/actions/TrackActions.jsx:56]: Avoid using {} when just string passed to attributes. id="se-hide-from-map-action"

File: map/src/menu/actions/TrackActions.jsx [hunk 6]
```diff
@@ -70,6 +71,7 @@ const TrackActions = forwardRef(({ track, setDisplayTrack, setOpenActions }, ref
                         </MenuItem>
                     ) : (
                         <MenuItem
+                            id={'se-show-on-map-action'}
```
Comment (RZR-UA) [inline map/src/menu/actions/TrackActions.jsx:74]: {'xxx'} => "xxx"

File: tests/selenium/src/tests/33-visible-tracks.mjs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    //check visible tracks
+    await clickBy(By.id(`se-actions-${trackName}`));
+    await waitBy(By.id('se-track-actions'));
+    await clickBy(By.id('se-show-on-map-action'));
+    await waitBy(By.id('se-visible-tracks-menu'));
+    await clickBy(By.id('se-visible-tracks-menu'));
+    await waitBy(By.id(`se-new-visible-track-${trackName}`));
+    await clickBy(By.id(`se-visible-switch-${trackName}`));
+    await waitBy(By.className(' leaflet-interactive'), { hidden: true });
```
Comment (RZR-UA) [inline tests/selenium/src/tests/33-visible-tracks.mjs:42]: Is this space in '_leaflet...' important?

**Review by RZR-UA — APPROVED** [review]

**Pull request merged**